### PR TITLE
Fix permalink local linkification to not strip via servers

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/15100

### CODE REVIEW

**Comment by turt2live** [comment]
Does this actually route through to the join though?

**Comment by t3chguy** [comment]
It gets passed to RoomView props.viaServers correctly so I'd wager so

File: src/utils/permalinks/Permalinks.js [hunk 1]
```diff
@@ -332,6 +332,10 @@ export function tryTransformPermalinkToLocalHref(permalink: string): string {
             if (permalinkParts.roomIdOrAlias) {
                 const eventIdPart = permalinkParts.eventId ? `/${permalinkParts.eventId}` : '';
                 permalink = `#/room/${permalinkParts.roomIdOrAlias}${eventIdPart}`;
+                if (permalinkParts.viaServers.length > 0) {
+                    const riotPermalinkConstructor = new SpecPermalinkConstructor();
```
Comment (turt2live) [inline src/utils/permalinks/Permalinks.js:336]: the variable name is wrong

**Review by turt2live — CHANGES_REQUESTED** [review]
it'd also be good to get a bit more of a confident answer for it working

File: src/utils/permalinks/Permalinks.js (hunk 1 above)
Comment (t3chguy) [reply src/utils/permalinks/Permalinks.js:336]: oh yeah, was gonna use the other c'tor but it needed a URL and the method I care about is identical in both

**Comment by t3chguy** [comment]
It doth;
image

**Review by turt2live — APPROVED** [review]

**Pull request merged**